packages/integration-test/test/integration.js

```diff
@@ -0,0 +1,393 @@
+/* global expect, web3, contract, it:true */
+const {
+    BurnProof,
+    DividendProof,
+    JoinSplitProof,
+    MintProof,
+    note,
+    PrivateRangeProof,
+    PublicRangeProof,
+    signer,
+    SwapProof,
+} = require('aztec.js');
+const bn128 = require('@aztec/bn128');
+
+const {
+    proofs: { BURN_PROOF, DIVIDEND_PROOF, JOIN_SPLIT_PROOF, SWAP_PROOF, MINT_PROOF, PRIVATE_RANGE_PROOF, PUBLIC_RANGE_PROOF },
+} = require('@aztec/dev-utils');
+const secp256k1 = require('@aztec/secp256k1');
+
+const BN = require('bn.js');
+const { getNotesForAccount, generateFactoryId } = require('../src/utils');
+const Setup = require('../src/setup');
+const config = require('../src/config');
+
+contract('Integration', (accounts) => {
+    let aztecAccount;
+    let ace;
+    let dividendValidator;
+    let erc20;
+    let joinSplitValidator;
+    let joinSplitFluidValidator;
+    let privateRangeValidator;
+    let publicRangeValidator;
+    let swapValidator;
+    let upgradeFactory;
+    let zkAsset;
+
+    let firstMintCounterValue;
+    let firstMintCounterNote;
+
+    let publicOwner;
+    let sender;
+    let opts;
+    let delegatedAddress;
+    let setup;
+
+    before(async () => {
+        setup = new Setup(config, accounts);
+
+        ({
+            ACE: ace,
+            Dividend: dividendValidator,
+            ERC20Mintable: erc20,
+            FactoryAdjustable201907: upgradeFactory,
+            JoinSplit: joinSplitValidator,
+            JoinSplitFluid: joinSplitFluidValidator,
+            PrivateRange: privateRangeValidator,
+            PublicRange: publicRangeValidator,
+            Swap: swapValidator,
+            ZkAssetAdjustable: zkAsset,
+        } = await setup.getContracts());
+
+        ({ sender, publicOwner, delegatedAddress, opts } = setup.getTransactionTestingAddresses());
```

Comment:
This is a good example of comment about making getters not do assignment: you've modified the instance your working with, is that expected?

Reply:
You're right, and you wouldn't expect this function to do that. Moved modifying the state to the constructor